app/controllers/tournaments_controller.rb

```diff
@@ -6,11 +8,31 @@ def index
   end
 
   def create
-    Tournament.create!(company_name: params[:company_name], start_date: params[:start_date], end_date: params[:end_date])
+    begin
+      new_tournament = Tournament.create!(company_name: params[:company_name], start_date: params[:start_date], end_date: params[:end_date])
+      results = params[:results]
+      results.each do |result|
+        new_user = User.create!(first_name: result[:firstName], last_name: result[:lastName], email: result[:email])
+        temp = Result.create!(user_id: new_user.id, tournament_id: new_tournament.id , answers: result[:answers])
+      end
+      render json: { success: "New Tournament Created" }, status: :ok 
+    rescue ActiveRecord::RecordInvalid => invalid
+      render json: {error: "Failed to create a new tournament, with error: #{invalid.record.errors.full_messages} - #{invalid.record.errors.details}"}, status: :internal_server_error
+    end
   end
 
   def show
-    Tournament.find(params[:id])
+    unless Tournament.exists?(params[:id])
```

Comment:
Instead of using `Tournament.find(params[:id])` which raises an error when not found, you can use `tournament = Tournament.find_by(id: params[:id])` which returns nil if it doesn't find anything.
Then you can check `if tournament.blank?` then return not found, and that way you only query for the tournament once instead of twice